### DISCUSSION TITLE
PDF access?

### DISCUSSION THREAD

**Discussion by stucka** [GSA-TTS/FAC#2633] [Q&A] [answered]
Is there a rough timeline for when we may get access to audit PDFs?

Is there an API endpoint that will get us a direct URL, or a predictable path built off report_id or something?

Thank you, and sorry!

Mike

**Comment by dbart00** [comment]
Hi, just wanted to echo Mike's comment---is there a way to access the uploaded pdfs via report_id?
I don't see any end points listed here:  https://api.fac.gov/dissemination/pdf/

Also, wanted to commend your teams work @jadudm. New incoming data is looking so much better. Much appreciation for front-end UEI validation.

**Reply by jadudm** [reply, marked as answer]
Thank you. I'll pass that on to the team. We have some more to do on the data side, but we're definitely going to keep pushing on quality of intake. 

That URL... if we have it somewhere, tell me where, and I'll fix it. That may have been a "for instance" kind of URL at some point, not meant to be "truth."

Give us a day or two to get docs together; the search feature is quietly live at this point. We missed our "by the end of the month" by a tiny bit. 

I'd recommend checking our updates page on the static site (https://www.fac.gov/info/updates/) at the start of next week. 

In the meantime, let me know what your experience is with

https://app.fac.gov/dissemination/report/pdf/<report-id>

I believe it should be the case that URLs to PDFs of reports will never change. So, you shouldn't have to scrape all of them (say) just for the sake of scraping. They should be available via browser or a `GET` request without too much fuss.

Yep. Confirmed. `curl <url> -L` (to follow the redirect) works just fine. It looks like `wget` just follows the redirect. What you have to do with your client/library of choice will vary. 

The underlying URL, BTW, times out after a minute. (Or, it should.) So, saving the redirect destination is *not* a good practice/not useful.

Again, docs to follow. This particular post may turn out to be inaccurate at some point in the indefinite, but likely far, future.

**Reply by dbart00** [reply]
Indeed, that was a "for instance" kind of URL--no need to fix anything.

I used this address your provided: 
https://app.fac.gov/dissemination/report/pdf/"report_id"
and it works perfectly 👍 got everything we needed for downstream pdf reporting requirement. 

Awesome thanks for letting me know!
